### PR TITLE
Release process doc

### PR DESCRIPTION
**Pull request by joshulyne**
#578

### CODE REVIEW

**Comment by integrii** [comment]
We should see what happens to our automatic container builds before we fully finish this document.

**Comment by chadbitzer** [comment]
Hold on merging this, I'm adding notes how Helm repo is publshed.

**Review by chadbitzer — APPROVED** [review]

File: docs/RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+6. Create a PR updating the project with the latest Kuberhealthy tag:
+    - [Makefile](../cmd/kuberhealthy/Makefile)
+    - Helm [values.yaml](../deploy/helm/kuberhealthy/values.yaml)
+    - Integration test values [values.yaml](../.ci/values.yaml)
+    - Once the PR is merged, make sure the Flat File Generation PR is also merged -- this updates the deployment flat spec files with the latest Kuberhealthy release image tag
+
+7. Write up and publish release notes, giving an update / latest changes from the last release: new *Kuberhealthy Checks* (if any), new *Features*, and *Bug Fixes*
+    - Tag the issue and pr as well as the contributor to the release notes
+
```
Comment (integrii) [inline docs/RELEASE.md:30]: What about creating a tag in github?  This says to publish notes, but does not talk about the tag that gets created.

File: docs/RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+5. Soak test in lab and prod for several days
+
+6. Create a PR updating the project with the latest Kuberhealthy tag:
+    - [Makefile](../cmd/kuberhealthy/Makefile)
+    - Helm [values.yaml](../deploy/helm/kuberhealthy/values.yaml)
+    - Integration test values [values.yaml](../.ci/values.yaml)
+    - Once the PR is merged, make sure the Flat File Generation PR is also merged -- this updates the deployment flat spec files with the latest Kuberhealthy release image tag
+
+7. Write up and publish release notes, giving an update / latest changes from the last release: new *Kuberhealthy Checks* (if any), new *Features*, and *Bug Fixes*
```
Comment (integrii) [inline docs/RELEASE.md:28]:
```suggestion
7. Write up release notes, giving an update / latest changes from the last release: new *Kuberhealthy Checks* (if any), new *Features*, and *Bug Fixes*.  Go into Github and create a new release with the appropriate version and change notes in the body.
```

**Review by integrii — CHANGES_REQUESTED** [review]

**Comment by integrii** [comment]
@joshulyne I think this updated README is valid given our new build processes and automation.  More work would be great to automate the tagging and pushing of versioned images, but this process should be accurate to current day.

File: docs/RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+To release a new version of Kuberhealthy:
+
+1. Ensure that all issues in the release milestone are completed
+1. Verify that the master branch has all the latest changes that have been tested and verified
+1. After any running `Build and Push kuberhealthy Latest` action is completed, test the image `kuberhealthy/kuberhealthy:latest` to validate the release
+1. If the release looks good, tag it with the new upcoming version and push it to docker hub:
+    ```
+    docker pull kuberhealthy/kuberhealthy:latest
+    docker tag kuberhealthy/kuberhealthy:latest kuberhealthy/kuberhealthy:v2.3.4
+    docker push kuberhealthy/kuberhealthy:v2.3.4
```
Comment (joshulyne) [inline docs/RELEASE.md:12]:
```suggestion
    docker tag kuberhealthy/kuberhealthy:latest kuberhealthy/kuberhealthy:{release_version}
    docker push kuberhealthy/kuberhealthy:{release_version}
```

Comment (joshulyne) [reply docs/RELEASE.md:12]: this is just a nitpick suggestion

**Pull request merged**